### ISSUE TITLE
Build errors on my arch system

### ISSUE THREAD

**Issue by janus-reith** [astoeckel/acinerella#2]
Output of cmake(no problems here):

```
-- The C compiler identification is GNU 5.3.0
-- The CXX compiler identification is GNU 5.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'libavcodec'
--   Found libavcodec, version 57.24.102
-- Checking for module 'libavformat'
--   Found libavformat, version 57.25.100
-- Checking for module 'libavutil'
--   Found libavutil, version 55.17.103
-- Checking for module 'libswscale'
--   Found libswscale, version 4.0.100
-- Checking for module 'libswresample'
--   Found libswresample, version 2.0.101
-- Found FFMPEG or Libav: /usr/lib64/libavcodec.so;/usr/lib64/libavformat.so;/usr/lib64/libavutil.so;/usr/lib64/libswscale.so;/usr/lib64/libswresample.so, /usr/include
-- Configuring done
-- Generating done
-- Build files have been written to: /home/janus/acinerella/build
```

Output of make:

```
Scanning dependencies of target acinerella
[ 25%] Building C object src/CMakeFiles/acinerella.dir/acinerella.c.o
/home/janus/acinerella/src/acinerella.c: In Funktion »ac_free_package«:
/home/janus/acinerella/src/acinerella.c:604:3: Warnung: »av_free_packet« ist veraltet [-Wdeprecated-declarations]
   av_free_packet(self->pPack);
   ^
In file included from /usr/include/libavformat/avformat.h:318:0,
                 from /home/janus/acinerella/src/acinerella.c:22:
/usr/include/libavcodec/avcodec.h:4040:6: Anmerkung: hier deklariert
 void av_free_packet(AVPacket *pkt);
      ^
/home/janus/acinerella/src/acinerella.c: Auf höchster Ebene:
/home/janus/acinerella/src/acinerella.c:614:6: Warnung: ISO-C verbietet Vorwärts-Referenzen auf »enum«-Typen [-Wpedantic]
 enum PixelFormat convert_pix_format(ac_output_format fmt)
      ^
/home/janus/acinerella/src/acinerella.c:614:18: Fehler: Rückgabetyp ist unvollständig
 enum PixelFormat convert_pix_format(ac_output_format fmt)
                  ^
/home/janus/acinerella/src/acinerella.c: In Funktion »convert_pix_format«:
/home/janus/acinerella/src/acinerella.c:618:11: Warnung: »return« mit Wert in void zurückgebender Funktion
    return AV_PIX_FMT_RGB24;
           ^
/home/janus/acinerella/src/acinerella.c:620:11: Warnung: »return« mit Wert in void zurückgebender Funktion
    return AV_PIX_FMT_BGR24;
           ^
/home/janus/acinerella/src/acinerella.c:622:11: Warnung: »return« mit Wert in void zurückgebender Funktion
    return AV_PIX_FMT_RGB32;
           ^
/home/janus/acinerella/src/acinerella.c:624:11: Warnung: »return« mit Wert in void zurückgebender Funktion
    return AV_PIX_FMT_BGR32;
           ^
/home/janus/acinerella/src/acinerella.c:626:9: Warnung: »return« mit Wert in void zurückgebender Funktion
  return AV_PIX_FMT_RGB24;
         ^
/home/janus/acinerella/src/acinerella.c: In Funktion »ac_create_video_decoder«:
/home/janus/acinerella/src/acinerella.c:664:2: Warnung: »avpicture_get_size« ist veraltet [-Wdeprecated-declarations]
  pDecoder->decoder.buffer_size = avpicture_get_size(
  ^
In file included from /usr/include/libavformat/avformat.h:318:0,
                 from /home/janus/acinerella/src/acinerella.c:22:
/usr/include/libavcodec/avcodec.h:4898:5: Anmerkung: hier deklariert
 int avpicture_get_size(enum AVPixelFormat pix_fmt, int width, int height);
     ^
/home/janus/acinerella/src/acinerella.c:666:6: Fehler: falsche Benutzung eines void-Ausdruckes
      pDecoder->pCodecCtx->width, pDecoder->pCodecCtx->height);
      ^
/home/janus/acinerella/src/acinerella.c:671:2: Warnung: »avpicture_fill« ist veraltet [-Wdeprecated-declarations]
  avpicture_fill((AVPicture *)(pDecoder->pFrameRGB),
  ^
In file included from /usr/include/libavformat/avformat.h:318:0,
                 from /home/janus/acinerella/src/acinerella.c:22:
/usr/include/libavcodec/avcodec.h:4883:5: Anmerkung: hier deklariert
 int avpicture_fill(AVPicture *picture, const uint8_t *ptr,
     ^
/home/janus/acinerella/src/acinerella.c:674:17: Fehler: falsche Benutzung eines void-Ausdruckes
                 pDecoder->pCodecCtx->width, pDecoder->pCodecCtx->height);
                 ^
/home/janus/acinerella/src/acinerella.c: In Funktion »ac_decode_video_package«:
/home/janus/acinerella/src/acinerella.c:781:7: Fehler: falsche Benutzung eines void-Ausdruckes
       SWS_BICUBIC, NULL, NULL, NULL);
       ^
src/CMakeFiles/acinerella.dir/build.make:62: die Regel für Ziel „src/CMakeFiles/acinerella.dir/acinerella.c.o“ scheiterte
make[2]: *** [src/CMakeFiles/acinerella.dir/acinerella.c.o] Fehler 1
CMakeFiles/Makefile2:87: die Regel für Ziel „src/CMakeFiles/acinerella.dir/all“ scheiterte
make[1]: *** [src/CMakeFiles/acinerella.dir/all] Fehler 2
Makefile:127: die Regel für Ziel „all“ scheiterte
make: *** [all] Fehler 2
```

What could be the issue here?
Incompatible version of libav/ffmpeg?

**Comment by astoeckel** [comment]
Thanks for reporting this issue.

Seems like the FFmpeg API has changed once again -- on Fedora 23 I still have the following older library versions

```
-- Checking for module 'libavcodec'
--   Found libavcodec, version 56.60.100
-- Checking for module 'libavformat'
--   Found libavformat, version 56.40.101
-- Checking for module 'libavutil'
--   Found libavutil, version 54.31.100
-- Checking for module 'libswscale'
--   Found libswscale, version 3.1.101
-- Checking for module 'libswresample'
--   Found libswresample, version 1.2.101
```

I'll install the recent version of ffmpeg on my machine and check whether I can reproduce/fix the problem.

**Comment by janus-reith** [comment]
Thanks for your efforts!

**Comment by astoeckel** [comment]
Should be fixed with commit 0c7bd815787c000d8f9cbdd3a1f91af0f6321abd

Can you confirm?

Some functions used by Acinerella are now marked as deprecated. However, I'm not removing calls to those as long as the replacements are not at least supported natively by Fedora.

**Comment by janus-reith** [comment]
Awesome thank you, the switch to AVPixelFormat fixed this :)
